### PR TITLE
Use EnvironmentHelper for database URL and install variables.

### LINKED ISSUE #132: initializeDatabaseConnectionVariables - Should use EnvironmentHelper as everywhere else, so retrieving environment variables is unified.
### PHP Version

8.1

### Shopware Version

6.4.70

### Expected behaviour

initializeDatabaseConnectionVariables of this template should use the EnvironmentHelper utility as all others do.

Even better would be if the `EnvironmentHelper` is a service that could be decorated.

### Actual behaviour

initializeDatabaseConnectionVariables is accessing $_SERVER etc. directly. However, the parent method does use `EnvironmentHelper` so this might result in different behavior.

### How to reproduce

Check code: 
https://github.com/shopware/production/blob/78819bc811ce0a27481b7143e916305243754e87/src/Kernel.php#L29
vs
https://github.com/shopware/platform/blob/v6.4.7.0/src/Core/Kernel.php#L376
and https://github.com/shopware/platform/blob/v6.4.7.0/src/Core/Kernel.php#L209

### PR DESCRIPTION
**Pull request by AndreasA**
Fixes: https://github.com/shopware/production/issues/132

### CODE REVIEW

File: bin/console [hunk 1]
```diff
@@ -53,11 +55,13 @@ if ($input->getFirstArgument() === 'system:install') {
...
 
-if (trim($_SERVER['DATABASE_URL'] ?? '') === '') {
+$databaseUrl = EnvironmentHelper::getVariable('DATABASE_URL', getenv('DATABASE_URL'));
+$install = EnvironmentHelper::getVariable('INSTALL');
+if (trim($databaseUrl ?? '') === '') {
     // fake DATABASE_URL
     $_SERVER['DATABASE_URL'] = Kernel::PLACEHOLDER_DATABASE_URL;
 } else {
-    if (!isset($_SERVER['INSTALL'])) {
+    if (null === $install) {
```
Comment (keulinho) [inline bin/console]:
```suggestion
    if (EnvironmentHelper::hasVariable('INSTALL')) {
```

File: src/Kernel.php [hunk 2]
```diff
@@ -26,11 +27,10 @@ public function __construct(
...
     protected function initializeDatabaseConnectionVariables(): void
     {
-        $url = $_ENV['DATABASE_URL']
-            ?? $_SERVER['DATABASE_URL']
-            ?? getenv('DATABASE_URL');
+        $url = EnvironmentHelper::getVariable('DATABASE_URL', getenv('DATABASE_URL'));
+        $install = EnvironmentHelper::getVariable('INSTALL');
 
-        if (isset($_SERVER['INSTALL']) || $url === self::PLACEHOLDER_DATABASE_URL) {
+        if (null !== $install || $url === self::PLACEHOLDER_DATABASE_URL) {
```
Comment (keulinho) [inline src/Kernel.php]:
```suggestion
        if (EnvironmentHelper::hasVariable('INSTALL') || $url === self::PLACEHOLDER_DATABASE_URL) {
```

**Review by keulinho — CHANGES_REQUESTED** [review]

File: bin/console (hunk 1 above)
Comment (AndreasA) [reply bin/console]: done. I also added the `!` which you forgot in the suggestion I think?

File: src/Kernel.php (hunk 2 above)
Comment (AndreasA) [reply src/Kernel.php]: done

**Review by keulinho — APPROVED** [review]

**Comment by AndreasA** [comment]
@keulinho can you re-check I also added the changes to `bin/ci` now. as it uses nearly the same logic as `bin/console`

**Review by keulinho — APPROVED** [review]

**Review by shyim — APPROVED** [review]

**Comment by shyim** [comment]
Hey,

the PR looks good. We are currently in releasing new version and don't merge new things into the template. I will import it when the new version is released.

**Comment by keulinho** [comment]
Hey @AndreasA thanks for your contribution! :tada: :blue_heart:

**Pull request closed without merging**